src/cockpit/beiboot.py

```diff
@@ -186,9 +194,19 @@ def __init__(self, router: Router, destination: str, args: argparse.Namespace):
         super().__init__(router)
 
     async def do_connect_transport(self) -> None:
+        # do we have user/password (Basic auth) from the login page?
+        auth = await self.router.request_authorization("*")
+        if auth is not None and auth.startswith('Basic '):
+            user, basic_password = base64.b64decode(auth.split(' ', 1)[1]).decode().split(':', 1)
+            logger.debug("got username %s and password from Basic auth", user)
+            ssh_opts = ['-l', user]
```

Comment:
I guess this overrides `user@` given as part of the host specification on the command line?

Reply:
Yes,  it does. AFAICS there can only ever be one or the other -- "client" mode doesn't have "Username:" and "Password:" fields (and thus no basic auth), and "standard" login page never supported adding a user name in the "Connect to:" field. My gut feeling is that "Username:" should win, but if you prefer I can add a check for `@` in `self.destination` and either handle it or error out with "make up your mind"

Reply:
There is a test for this now, so resolving.